spring-integration-mqtt/src/main/java/org/springframework/integration/mqtt/inbound/Mqttv5PahoMessageDrivenChannelAdapter.java

```diff
@@ -162,28 +163,24 @@ protected void onInit() {
 	@Override
 	protected void doStart() {
 		ApplicationEventPublisher applicationEventPublisher = getApplicationEventPublisher();
-		this.topicLock.lock();
-		String[] topics = getTopic();
 		try {
 			this.mqttClient.connect(this.connectionOptions).waitForCompletion(getCompletionTimeout());
-			if (topics.length > 0) {
-				int[] requestedQos = getQos();
-				this.mqttClient.subscribe(topics, requestedQos).waitForCompletion(getCompletionTimeout());
-				String message = "Connected and subscribed to " + Arrays.toString(topics);
-				logger.debug(message);
-				if (applicationEventPublisher != null) {
-					applicationEventPublisher.publishEvent(new MqttSubscribedEvent(this, message));
-				}
-			}
 		}
 		catch (MqttException ex) {
-			if (applicationEventPublisher != null) {
-				applicationEventPublisher.publishEvent(new MqttConnectionFailedEvent(this, ex));
+			if (this.connectionOptions.isAutomaticReconnect()) {
```

Comment:
This one is a bit different.
The message driven channel adapter is really an active one and does its stuff automatycally - it produces messages whenever they are available on the MQTT topic(s).
So, I guess the logic you did here is OK: 
1. If we connected and automatic reconnect is set, then we are good to go.
2. If we didn't connect, we must reconnect, but if there is no automatic reconnect, we cannot guarantee that it will reconnect on that async `reconnect()`, so we indeed should just throw that exception and let end-user to control it from the outside.
3. If it us auto, then we call `reconnect()` and it has to be enough for eventual success connection.

Reply:
I'm not sure why we need to call `reconnect()` when we are not using auto

Reply:
Yeah... I see your point: let's handle that as a `MqttConnectionFailedEvent` instead!

In other words: I agree with what you have so far.


Sorry for a noise!